File: Csharp/games/numberguessing/numberguessing.cs
using System;

class Program
{
    static void Main()
    {
        Random random = new Random();
        int number = random.Next(1, 101); // Zufallszahl zwischen 1 und 100
        int guess = 0;
        int attempts = 0;

        Console.WriteLine("Willkommen zum Zahlenratespiel!");
        Console.WriteLine("Ich habe mir eine Zahl zwischen 1 und 100 ausgedacht. Versuche sie zu erraten!");

        // Spielschleife
        while (guess != number)
        {
            Console.Write("Gib deine Schätzung ein: ");
            string input = Console.ReadLine();

            // Eingabe validieren
            if (int.TryParse(input, out guess))
            {
                attempts++; // Zähler für Versuche erhöhen

                if (guess < number)
                {
                    Console.WriteLine("Zu niedrig! Versuch es noch einmal.");
                }
                else if (guess > number)
                {
                    Console.WriteLine("Zu hoch! Versuch es noch einmal.");
                }
                else
                {
                    Console.WriteLine($"Herzlichen Glückwunsch! Du hast die Zahl {number} in {attempts} Versuchen erraten.");
                }
            }
            else
            {
                Console.WriteLine("Bitte gib eine gültige Zahl ein.");
            }
        }
    }
}

File: Kotlin/games/numberguessing/numberguessing.kt
import kotlin.random.Random

fun main() {
    // Zufallszahl zwischen 1 und 100 generieren
    val number = Random.nextInt(1, 101)
    var guess: Int
    var attempts = 0

    println("Willkommen zum Zahlenratespiel!")
    println("Ich habe mir eine Zahl zwischen 1 und 100 ausgedacht. Versuche sie zu erraten!")

    // Spielschleife
    do {
        println("Gib deine Schätzung ein:")
        val input = readLine()

        // Überprüfen, ob die Eingabe eine gültige Zahl ist
        try {
            guess = input?.toInt() ?: throw NumberFormatException()
            attempts++  // Zähler für Versuche erhöhen

            if (guess < number) {
                println("Zu niedrig! Versuch es noch einmal.")
            } else if (guess > number) {
                println("Zu hoch! Versuch es noch einmal.")
            } else {
                println("Herzlichen Glückwunsch! Du hast die Zahl $number in $attempts Versuchen erraten.")
            }
        } catch (e: Exception) {
            println("Bitte gib eine gültige Zahl ein.")
        }
    } while (guess != number)
}
